tests/core/test_desktopfile.cpp

```diff
@@ -0,0 +1,210 @@
+// library headers
+#include <gtest/gtest.h>
+#include <boost/filesystem.hpp>
+
+// local headers
+#include "linuxdeploy/core/desktopfile.h"
+#include "../../src/core/desktopfilereader.h"
+
+using boost::bad_lexical_cast;
+using namespace linuxdeploy::core::desktopfile;
+
+namespace bf = boost::filesystem;
+
+class DesktopFileFixture : public ::testing::Test {
+public:
+    std::string testType;
+    std::string testName;
+    std::string testExec;
+    std::string testIcon;
+    std::string testDesktopFile;
+
+private:
+    void SetUp() override {
+        testType = "Application";
+        testName = "Simple Application";
+        testExec = "simple_app";
+        testIcon = "simple_app";
+
+        std::stringstream ss;
+        ss << "[Desktop Entry]" << std::endl
```

Comment:
This is a really simple desktop file. I would recommend using something more complex.
Like having also arguments at the exec entry. Localized entries and so on.

Reply:
There are no checks applied at all. We don't use any localization in linuxdeploy, so there's no code in that direction. All code is covered by unit tests. There's no reason to test for this unless we have functionality for that in the code.

Also, the simple app's desktop file is parsed.

Reply:
Added some more complex example to the desktop file reader. I think it's all covered now. You may add test cases in the future if you want to. This is the wrong place for such tests though, parsing is done in the `DesktopFileReader`.